File: app/src/main/java/com/cn29/aac/datasource/feedentry/db/FeedEntryDAO.java
package com.cn29.aac.datasource.feedentry.db;

import com.cn29.aac.repo.feedentry.FeedEntry;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

/**
 * Created by Charles Ng on 5/9/2017.
 */

@Dao
public interface FeedEntryDAO {

  @Query("SELECT * FROM feedEntrys")
  LiveData<List<FeedEntry>> getAll();

  @Query("SELECT * FROM feedEntrys WHERE uid = :uid LIMIT 1")
  LiveData<FeedEntry> findByUid(int uid);

  @Insert
  void insertAll(FeedEntry... feedEntries);

  @Delete
  void delete(FeedEntry feedEntry);

  @Update
  int update(FeedEntry feedEntry);

}

File: app/src/main/java/com/cn29/aac/datasource/feedentry/db/FeedEntryDAO.kt
package com.cn29.aac.datasource.feedentry.db

import androidx.lifecycle.LiveData
import androidx.room.*
import com.cn29.aac.repo.feedentry.FeedEntry

@Dao
interface FeedEntryDAO {
    @Query("SELECT * FROM feedEntrys")
    fun getAll(): LiveData<List<FeedEntry>>

    @Query("SELECT * FROM feedEntrys WHERE uid = :uid LIMIT 1")
    fun findByUid(uid: Int): LiveData<FeedEntry>

    @Insert
    fun insertAll(vararg feedEntries: FeedEntry?): List<Long>

    @Delete
    fun delete(feedEntry: FeedEntry?): Int?

    @Update
    fun update(feedEntry: FeedEntry?): Int?
}